### PR TITLE
dynamic_forward_proxy: envoy.reloadable_features.enable_dns_cache_circuit_breakers deprecation

### LINKED ISSUE #14642: envoy.reloadable_features.enable_dns_cache_circuit_breakers deprecation
Your change #11028 (dynamic_forward_proxy: DNS Cache circuit breaker) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Shikugawa <rei@tetrate.io>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: fix #14642. remove `envoy.reloadable_features.enable_dns_cache_circuit_breakers` runtime flag.
Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: api/envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto [hunk 1]
```diff
@@ -92,7 +92,6 @@ message DnsCacheConfig {
   config.cluster.v3.Cluster.RefreshRate dns_failure_refresh_rate = 6;
 
   // The config of circuit breakers for resolver. It provides a configurable threshold.
-  // If `envoy.reloadable_features.enable_dns_cache_circuit_breakers` is enabled,
   // envoy will use dns cache circuit breakers with default settings even if this value is not set.
```
Comment (adisuissa) [inline api/envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto:95]: nit: `Envoy` instead of `envoy`

**Review by adisuissa — COMMENTED** [review]
API nit comment.
Thanks!

File: source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc [hunk 2]
```diff
@@ -71,19 +71,9 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
...
 
-  const bool should_use_dns_cache_circuit_breakers =
-      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_dns_cache_circuit_breakers");
-
-  circuit_breaker_ = config_->cache().canCreateDnsRequest(
-      !should_use_dns_cache_circuit_breakers
-          ? absl::make_optional(std::reference_wrapper<ResourceLimit>(
-                cluster_info_->resourceManager(route_entry->priority()).pendingRequests()))
-          : absl::nullopt);
+  circuit_breaker_ = config_->cache().canCreateDnsRequest();
```
Comment (asraa) [inline source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc:74]: drive-by, can you remove the un-needed runtime_features include and dep please?

File: test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_test.cc [hunk 3]
```diff
@@ -90,7 +90,7 @@ TEST_F(ProxyFilterTest, HttpDefaultPort) {
 
   EXPECT_CALL(callbacks_, route());
   EXPECT_CALL(cm_, getThreadLocalCluster(_));
-  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_(_))
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_())
```
Comment (asraa) [inline test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_test.cc:93]: ScopedRuntime is no longer needed in this test, I think.

**Comment by htuch** [comment]
LGTM API modulo @adisuissa  comment.

File: test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_test.cc (hunk 3 above)
Comment (Shikugawa) [reply test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_test.cc:93]: fixed

File: source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc (hunk 2 above)
Comment (Shikugawa) [reply source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc:72]: fixed

**Review by htuch — COMMENTED** [review]
/lgtm api

**Review by asraa — APPROVED** [review]
Thanks!
